static/panes/ssa-view.ts

```diff
@@ -0,0 +1,838 @@
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+import {Pane} from './pane.js';
+import * as monaco from 'monaco-editor';
+import $ from 'jquery';
+import _ from 'underscore';
+import * as fileSaver from 'file-saver';
+
+import {CfgState} from './cfg-view.interfaces.js';
+import {Hub} from '../hub.js';
+import {Container} from 'golden-layout';
+import {PaneState} from './pane.interfaces.js';
+import * as utils from '../utils.js';
+
+import {CFGResult} from '../../types/compilation/cfg.interfaces.js';
+import {GraphLayoutCore} from '../graph-layout-core.js';
+import * as MonacoConfig from '../monaco-config.js';
+import TomSelect from 'tom-select';
+import {assert, unwrap} from '../assert.js';
+import {CompilationResult} from '../compilation/compilation.interfaces.js';
+import {CompilerInfo} from '../compiler.interfaces.js';
+import {escapeHTML} from '../../shared/common-utils.js';
+
+const ColorTable = {
+    red: '#FE5D5D',
+    green: '#76E381',
+    blue: '#65B7F6',
+    grey: '#c5c5c5',
+};
+
+type Coordinate = {
+    x: number;
+    y: number;
+};
+
+const DZOOM = 0.1;
+const MINZOOM = 0.1;
+
+const EST_COMPRESSION_RATIO = 0.022;
+
+function attrs(attributes: Record<string, string | number | null>) {
+    return Object.entries(attributes)
+        .map(([k, v]) => `${k}="${v}"`)
+        .join(' ');
+}
+
+function special_round(x: number) {
+    assert(x >= 0);
+    if (x === 0) {
+        return 0;
+    }
+    const p = Math.pow(10, Math.floor(Math.log10(x)));
+    // prettier-ignore
+    const candidates = [
+        Math.round(x / p) * p - p / 2,
+        Math.round(x / p) * p,
+        Math.round(x / p) * p + p / 2,
+    ];
+    return Math.trunc(candidates.sort((a, b) => Math.abs(x - a) - Math.abs(x - b))[0]);
+}
+
+function size_to_human(bytes: number) {
+    if (bytes < 1000) {
+        return special_round(bytes) + ' B';
+    } else if (bytes < 1_000_000) {
+        return special_round(bytes / 1_000) + ' KB';
+    } else if (bytes < 1_000_000_000) {
+        return special_round(bytes / 1_000_000) + ' MB';
+    } else {
+        return special_round(bytes / 1_000_000_000) + ' GB';
+    }
+}
+
+export class Ssa extends Pane<CfgState> {
```

Comment:
This file appears to be mostly identical to cfg-view.ts with the main difference appearing to be the parseSSA function. Most of the CFG parsing is done on the back-end, see [lib/cfg/cfg-parsers](https://github.com/compiler-explorer/compiler-explorer/tree/main/lib/cfg/cfg-parsers). I would recommend reworking your implementation to parse a control flow graph for gcc IR output on the back-end and simply use the existing CFG view pane.

Reply:
@adityaagre you didn't appear to reply to @jeremy-rifkin 's observation here